src/app/content/highlights/components/Highlights.tsx

```diff
@@ -134,3 +206,145 @@ export const SectionHighlights = ({ location, highlights }: SectionHighlightsPro
     </React.Fragment>
   );
 };
+
+interface HighlightToggleEditProps {
+  color: HighlightColorEnum;
+  onDelete: () => void;
+  onEdit: () => void;
+  onColorChange: (color: string) => void;
+}
+
+// tslint:disable-next-line:variable-name
+const HighlightToggleEdit = ({
+  color,
+  onColorChange,
+  onEdit,
+  onDelete,
+}: HighlightToggleEditProps) => <Styled.HighlightToggleEdit>
+  <Dropdown
+    toggle={<MenuToggle/>}
+    className='highlight-toggle-edit'
+  >
+    <Styled.HighlightToggleEditContent>
+      <ColorPicker
+        color={color}
+        onChange={onColorChange}
+        onRemove={() => onDelete()}
+      />
+      <DropdownList>
+        <DropdownItem
+          message='i18n:highlighting:dropdown:edit'
+          prefix={<EditIcon/>}
+          onClick={() => onEdit()}
+        />
+        <DropdownItem
+          message='i18n:highlighting:dropdown:delete'
+          prefix={<TrashAltIcon/>}
+          onClick={() => onDelete()}
+        />
+      </DropdownList>
+    </Styled.HighlightToggleEditContent>
+  </Dropdown>
+</Styled.HighlightToggleEdit>;
+
+interface HighlightAnnotationProps {
+  annotation: string;
+  isEditable: boolean;
+  onSave: (annotation: string) => void;
+  onCancel: () => void;
+}
+
+// tslint:disable-next-line:variable-name
+const HighlightAnnotation = (
+  { annotation, isEditable, onSave, onCancel }: HighlightAnnotationProps
+) => {
+  const [anno, setAnno] = React.useState(annotation);
+
+  const onKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      onCancel();
+      return;
+    } else if (e.key === 'Enter' && !e.shiftKey) {
+      onSave(anno);
```

Comment:
if this isn't in the spec (i don't think it is?) please don't do any special enter key handling

Reply:
ok, sorry, I'm just used to working without design specs when review is ending at: please implement also this, this and this :D